docs/src/pages/components/tree-view/ControlledTreeView.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import TreeView from '@material-ui/lab/TreeView';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import TreeItem from '@material-ui/lab/TreeItem';
+
+const useStyles = makeStyles({
+  root: {
+    height: 216,
+    flexGrow: 1,
+    maxWidth: 400,
+  },
+});
+
+export default function ControlledTreeView() {
+  const classes = useStyles();
+  const [expanded, setExpanded] = React.useState<string[]>([]);
+
+  const handleChange = (nodes: string[], expand: boolean) => {
```

Comment:
What are thoughts on changing the `onNodeToggle` behavior to allow direct mapping? Something like this:
```jsx
const handleChange = (event, newExpanded) => {
  setExpanded(newExpanded);
}
```

The point that makes me wonder is that the related issue emphasize is on the `expanded` prop more than the `onNodeToggle` callback.

Reply:
What happens with collapses? Wouldn’t the change depend on state when controlling the tree view?

Reply:
What do you mean by collapses? Should we consider, by exclusion that if a node is not in the expanded list, he's collapsed?

Reply:
True, it is possible. I’ll see what it looks like after. I’m not sure how easy populating the event will be.

Reply:
My hope is that it would simplify the code of the tree view.